router/core/src/main/scala/io/buoyant/router/RoutingFactory.scala

```diff
@@ -165,7 +167,11 @@ class RoutingFactory[Req, Rsp](
             Future.exception(UnknownDst(req0, unidentified.reason))
           case Throw(e) =>
             Annotations.Failure.Identification.record(e.getMessage)
-            Future.exception(UnknownDst(req0, e.getMessage))
+            e match {
+              case e: ResponseException[Rsp] => Future.exception(e)
```

Comment:
This lead my down a bit of a rabbit hole trying to figure out how Monitors work but I think we can have the ResponseException extend `com.twitter.logging.HasLogLevel` and set the log level to TRACE or something like that.